react-ui/src/components/addOrder/OrderReferralRequest.js

```diff
@@ -0,0 +1,142 @@
+    Select,
+    SelectItem,
+    Table,
+    TableBody,
+    TableCell,
+    TableHead,
+    TableHeader,
+    TableRow,
+    TextInput
+} from "@carbon/react";
+import {UserInformationContext} from "../layout/Layout";
+
+
+const header = [{key: 'reason', header: 'Reason for Referral*'}, {key: 'referrer', header: 'Referrer'}, {
+    key: 'institute', header: 'Institute'
+}, {key: '', header: 'Sent Date\n' + '(dd/mm/yyyy)'}, {key: 'name', header: 'Test Name'},];
+
+const OrderReferralRequest = ({index, selectedTests, referralReasons, referralOrganizations}) => {
+    const {user} = useContext(UserInformationContext);
+    const [referralRows, setReferralRows] = useState([]);
+
+    useEffect(() => {
+        const rows = [];
+        selectedTests.map(test => {
+            let elementId = index + "_" + test.id;
+            let row = {
+                reason: <ReferralReason index={elementId} reasons={referralReasons}/>,
+                referrer: <Referrer index={elementId} user={user}/>,
+                institute: <Institute index={elementId} organizations={referralOrganizations}/>,
+                sentDate: <SentDate index={elementId}/>,
+                testName: <TestName index={elementId} test={test}/>
+            }
+            rows.push(row);
+        });
+        setReferralRows(rows);
+
+    }, [selectedTests]);
+
+
+    const ReferralReason = ({index, reasons}) => {
+        return (<>
+            <Select
+                labelText=""
+                id={`referralReasonId_` + index}>
+                {
+                    reasons.map(reason => {
+                        return (
+                            <SelectItem
+                                key={reason.id}
+                                text={reason.value}
+                                value={reason.id}
+                            />
+                        );
+                    })
+                }
+            </Select>
+        </>);
+    };
+    const Referrer = ({index, user}) => {
+        return (<>
+            <TextInput id={`referrer_${index}`} labelText="" value={user.firstName + " " + user.lastName}/>
+        </>);
+    };
+
+    const Institute = ({index, organizations}) => {
+        return (<>
+            <Select
+                labelText="" id={`referredInstituteId_` + index}>
+                {
+                    organizations.map(org => {
+                        return (
+                            <SelectItem
+                                key={org.id}
+                                text={org.value}
+                                value={org.id}
+                            />
+                        );
+                    })
+                }
+            </Select>
+        </>);
+    };
+
+    const SentDate = ({index}) => {
+        return (<>
+            <DatePicker dateFormat="m/d/Y" datePickerType="single">
+                <DatePickerInput id={`sendDate_` + index} placeholder="dd/mm/yyyy" type="text" labelText=""/>
```

Comment:
we will need the date format to be configurable based on the Site Information

Reply:
Do we have an endpoint that queries site configurations?